walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,97 @@
 
 package com.google.sps;
 
+import java.io.*;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.stream.*;
+
 
 public final class FindMeetingQuery {
+
+  /** Returns a Collection of time ranges when meeting {@code request} can be scheduled in the day of 
+    * events so that all attendees are free 
+    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      // if the meeting lasts more than a day, there is no solution
+      return new ArrayList<TimeRange>();
+    }
+
+    // array that will represent the number of meetings happening during that minute
+    ArrayList<Integer> meetings = new ArrayList<Integer>();
+    // before processing the events, the number of meetings for each minute is 0
+    meetings.addAll(Collections.nCopies(TimeRange.END_OF_DAY - TimeRange.START_OF_DAY + 1, 0));
```

Comment:
nit: I'm not even sure you need START_OF_DAY + 1 here or is START_OF_DAY sufficient.

IIUC, using START_OF_DAY + 1 would allow you to consider meetings of duration 0 (starts and ends in the last minute), but not sure if that's even a valid usecase.